Add instructions to build in Docker

T-vK opened the pull request:
I cleaned the instructions up a bit and added instructions on how to build AsteroidOS in Docker.

FlorentRevest commented on pages/wiki/building-asteroidos.hbs:
```diff
@@ -10,35 +10,91 @@ layout: documentation
   <h1 id="general-information">General information</h1>
 </div>
-These instructions have been tested on Ubuntu 19.04, but should also be applicable to Debian Sid (at least at the time of writing). For other distributions the dependencies may have different names and you may have to install additional ones.
+<p>These instructions have been tested on Ubuntu 19.04, but should also be applicable to Debian Sid (at least at the time of writing). For other distributions the dependencies may have different names and you may have to install additional ones.</p>
```
You should move this paragraph to inside the "build without docker" section

FlorentRevest replied:
You should also add an introduction to why one could potentially prefer the Docker build over a build without Docker.

FlorentRevest commented on pages/wiki/building-asteroidos.hbs:
```diff
@@ -10,35 +10,91 @@ layout: documentation
+<p>Create a Docker container called <em>asteroidos-toolchain</em> from the Docker image which we also called <em>asteroidos-toolchain</em> and build AsteroidOS for <em>dory</em> (the LG G Watch) within the container:</p>
+<pre><code>sudo docker rm -f asteroidos-toolchain ; sudo docker run --name asteroidos-toolchain -it -v /etc/passwd:/etc/passwd -u `id -u`:`id -g` -v "$HOME/.gitconfig:/$HOME/.gitconfig" -v "$(pwd):/asteroid" asteroidos-toolchain bash -c "source ./prepare-build.sh dory && bitbake asteroid-image"</code></pre>
+
+<p>Note: The files created during the build simply go to the current direcotry (which should be <em>asteroid</em>), more spcifically, the output goes to the subdirectories <em>asteroid/src</em> and <em>asteroid/build</em>.</p>
```
directory*

FlorentRevest replied:
specifically*

FlorentRevest commented on pages/wiki/building-asteroidos.hbs:
```diff
@@ -10,35 +10,91 @@ layout: documentation
+
+
+<div class="page-header">
+  <h3 id="docker-run-explained">What exactly does that long <em>docker rm ... ; docker run ...</em> command do?</h3>
```
I don't think we should have a separate section for that, you can just detail it in the above section
+ the wording of "What exactly does that long command do?" is not the wording I would expect to find in a documentation

T-vK replied:
Okay, sure I'll change it. What did you have in mind instead of `What exactly does that long <em>docker rm ... ; docker run ...</em> command do?`?  
(It sounds pretty okay to me to be honest.)

Would `Explanation of the above <em>docker rm ... ; docker run ...</em> command:` be better?

FlorentRevest replied:
Yes it would be better

FlorentRevest commented on pages/wiki/building-asteroidos.hbs:
```diff
@@ -10,35 +10,91 @@ layout: documentation
+
+# bash -c "source ./prepare-build.sh dory && bitbake asteroid-image"   This is the command to be executed inside of the container.
+                                                                       "dory" being the codename of your watch.
+                                                                       (You have to use source and not just "." here!)</code></pre>
```
I don't think it's necessary to tell users not to run a command that is not even given

T-vK replied:
If you don't like it, I'll remove. But I thought it would be a good idea to mention it because in other places of the documentation `. ./prepare-build.sh` is used and it wasn't obvious to me that the `.` doesn't behave the same as `source` within a `bash -c "..."` call. For instance I could see please copying `. ./prepare-build.sh dory` and `bitbake meta-toolchain-qt5` from the `Creating an Asteroid App`-page.

FlorentRevest replied:
Let's just replace . with source in the couple of remaining places. Half of the users miss the dot and complain that it doesn't work anyway.

FlorentRevest commented on pages/wiki/building-asteroidos.hbs:
```diff
@@ -10,35 +10,91 @@ layout: documentation
 </div>
-<p>After a while, the generated image should be available in <em>build/tmp-glibc/deploy/images/dory/</em>.</p> Install AsteroidOS using your usual device's instructions.
+<p>After a while, the generated image should be available in <em>build/tmp-glibc/deploy/images/dory/</em>. (No matter if you built with or without Docker.)</p>
+<p>Install AsteroidOS using your usual device's instructions, which you can find when you go to the <a href="{{rel 'install'}}">Install page</a> and click your device there.</p>
```
"which you can find when you go to the Install page and click your device there"
->
"which you can find on the Install page."

FlorentRevest commented on pages/wiki/building-asteroidos.hbs:
```diff
@@ -10,35 +10,91 @@ layout: documentation
+<pre><code>git clone https://github.com/AsteroidOS/asteroid.git
+cd asteroid/</code></pre>
+
+<p>If you haven't set a global git name and email yet, please do so, otherwise the build will not succeed! For example:</p>
```
the "please do so, otherwise the build will not succeed!" is not a wording that should be found in a documentation. You should keep a neutral tone and it is implied that the instructions you give are necessary for the build to succeed, otherwise you wouldn't give them on the first place.
I would suggest "If you haven't set a global git name and email yet, adapt the following commands to your information." If you want you can also justify this with something like "This is only required to clone some git repositories when building" but no exclamation mark is necessary.

FlorentRevest commented on pages/wiki/building-asteroidos.hbs:
```diff
@@ -10,35 +10,91 @@ layout: documentation
 </code></pre>
-<p>This repository only contains a shell script that populates <em>src/</em> with OpenEmbedded and the appropriate Asteroid layers. Then, it setups the environment for a bitbake build. The following command will setup a build for <em>dory</em> (the LG G Watch) but you can also build an image for other watches by using the corresponding codename.</em></p>
+
+<p>If you haven't set a global git name and email yet, please do so, otherwise the build will not succeed! For example:</p>
```
See below for a comment on the same sentence.

FlorentRevest commented on pages/wiki/building-asteroidos.hbs:
```diff
@@ -10,35 +10,91 @@ layout: documentation
+
+<p>First install Docker and Git on your system if you haven't already, then clone the main repository using the following command:</p>
+<pre><code>git clone https://github.com/AsteroidOS/asteroid.git
+cd asteroid/</code></pre>
```
Could you factorize the git clone part out of the with and without docker instructions? Since this is a shared first step

FlorentRevest commented on pages/wiki/building-asteroidos.hbs:
```diff
@@ -10,35 +10,91 @@ layout: documentation
+<p>If you haven't set a global git name and email yet, please do so, otherwise the build will not succeed! For example:</p>
+
+<pre><code>git config --global user.email "you@example.com"
+git config --global user.name "Your Name"</code></pre>
```
You could actually also factorize those git commands out of the with/without docker sections, since it is also common.

T-vK commented:
Okay, I think I have adjusted everything you commented on. I also added `requires a lot of disk space (potentially more than 100GB)` which I think would be very good to know for everyone before attempting to build.

FlorentRevest commented:
The paragraph about Docker is a bit too dense to my taste but that will do.
Thank you ;)

T-vK commented:
No problem, you're welcome. :)
